engine/src/main/java/io/zeebe/engine/processor/workflow/deployment/model/element/ExecutableLoopCharacteristics.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.deployment.model.element;
+
+import static io.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.zeebe.msgpack.jsonpath.JsonPathQuery;
+import java.util.Optional;
+import org.agrona.DirectBuffer;
+
+public class ExecutableLoopCharacteristics {
+
+  private final boolean isSequential;
+
+  private final JsonPathQuery inputCollection;
+  private final Optional<DirectBuffer> inputElement;
```

Comment:
I like the use of `Optional`, but afaik, it's meant primarily for return types or wrapping function calls, and not for class attributes (e.g. [Brian Goetz's SO answer](https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555))

Could we instead just wrap it when `getInputElement` is called, whether as the return type or just wrapping the call itself?